lib/DynamicCreative.js

```diff
@@ -1,42 +1,54 @@
 
 class DynamicCreative {
 
-  constructor() {
+  constructor(canvas) {
     // TODO implement this
+    this._canvas = canvas;
+    this._elements = [];
+    this._aspects = [];
   }
 
   add(elements) {
     // TODO implement this
+    if(Array.isArray(elements)) {
+      this._elements = elements;
+      this._elements.map(item => {
+        let rate = (item.w/item.h).toFixed(1);
+        this._aspects.push(rate);
+      });
+    }
   }
 
   toParams(canvas) {
     // TODO implement this
-    return [
-      { id: 'image1',
-        type: 'Image',
-        x: 24,
-        y: 30,
-        w: 192,
-        h: 106.66666666666667 },
-      { id: 'image2', type: 'Image', x: 180, y: 160, w: 48, h: 24 },
-      { id: 'text1',
-        type: 'Text',
-        x: 20,
-        y: 146.66666666666669,
-        w: 150,
-        h: 128 },
-      { id: 'text2',
-        type: 'Text',
-        x: 24,
-        y: 284.6666666666667,
-        w: 192,
-        h: 45 },
-    ]
+    let scalecanvas = canvas;
+    let newElements = [];
+    let originalElements = this._elements;
+    let containerWidth = canvas.w;
+    let boxSpacing = 15;
+    let layoutGeometry = require('justified-layout')(this._aspects, {
```

Comment:
it is good manner to search for third-party packages. but you still ignore the real requirement and background, etc. the results in the debug page are far from good.